bib/cmd/bootc-image-builder/main.go

```diff
@@ -313,16 +313,45 @@ func cmdBuild(cmd *cobra.Command, args []string) error {
 		if !slices.Contains(imgTypes, "ami") {
 			return fmt.Errorf("aws flags set for non-ami image type (type is set to %s)", strings.Join(imgTypes, ","))
 		}
-		// initialise the client to check if the env vars exist before building the image
+
+		// check as many permission prerequisites as possible before starting
 		client, err := awscloud.NewDefault(region)
 		if err != nil {
 			return err
 		}
 
-		fmt.Printf("Checking AWS permission by listing regions...\n")
-		if _, err := client.Regions(); err != nil {
+		fmt.Printf("Checking AWS region access...\n")
+		var regions []string
+		regions, err = client.Regions()
+		if err != nil {
+			return fmt.Errorf("retrieving AWS regions for '%s' failed: %w", region, err)
+		}
+
+		if !slices.Contains(regions, region) {
+			return fmt.Errorf("given AWS region '%s' not found", region)
+		}
+
+		fmt.Printf("Checking AWS bucket...\n")
```

Comment:
Minor/optional and not new, but this could be `fmt.Println()` and avoid the `\n`.

(Rust of course got this right, the `println!()` macro is saner...)